docs/src/pages/components/data-grid/components/CustomHeader.tsx

```diff
@@ -0,0 +1,67 @@
+import { makeStyles } from '@material-ui/core/styles';
+import Pagination from '@material-ui/lab/Pagination';
+import { useDemoData } from '@material-ui/x-grid-data-generator';
+import * as React from 'react';
+import {
+  useGridSlotComponentProps,
+  DataGrid,
+  GridHeader,
+} from '@material-ui/data-grid';
+
+export function PaginationComponent(props: { color?: 'primary' }) {
+  const { state, apiRef } = useGridSlotComponentProps();
+  return (
+    <Pagination
+      className="my-custom-pagination"
+      page={state.pagination.page + 1}
+      color={props.color}
+      count={state.pagination.pageCount}
+      onChange={(event, value) => apiRef.current.setPage(value)}
```

Comment:
DataGrid is not allowed to use apiRef. This looks like a design flaw in the component structure. Shouldn't the pagination slot receive the props that he needs to render it? But maybe showing the usage of GridPagination in this demo would be enough?

Reply:
removed